claim.go

```diff
@@ -101,15 +101,28 @@ func NewSchemaHashFromHex(s string) (SchemaHash, error) {
 		return SchemaHash{}, err
 	}
 
-	if len(schemaEncodedBytes) != len(schemaHash) {
+	if len(schemaEncodedBytes) > len(schemaHash) {
```

Comment:
I have commented already about checking the length of the input hex string. If you work with sha256, for example, you expect the length to be 32 bytes. Not less, not more. Because if you get fewer bytes, what is the rule to padding — append zeros from the left, or from the right? You can't just use this function, you need to read how it is implemented to understand how it is working.

Reply:
Roll-back after discussion